### PR TITLE
CLOUDP-82831: Improve argument validation

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-82831

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

This relates to a recent bug report where the users was trying to use emails instead of ids so adding a little validation here and there

### CODE REVIEW

File: internal/cli/require/require.go [hunk 1]
```diff
@@ -49,6 +50,17 @@ func ExactArgs(n int) cobra.PositionalArgs {
...
+// any of those args is not an ObjectID.
+func ExactObjectIDArgs(n int) cobra.PositionalArgs {
+	return func(cmd *cobra.Command, args []string) error {
+		if err := ExactArgs(n)(cmd, args); err != nil {
+			return err
+		}
+		return objectIDArgs(args)
+	}
+}
+
```
Comment (andreaangiolillo) [inline internal/cli/require/require.go:63]: [q] Could this function takes `func objectIDArgs()` as input? so we won't have to define the same method if we want to apply other validations to the args.

File: internal/cli/require/require.go [hunk 2]
```diff
@@ -81,6 +93,38 @@ func MinimumNArgs(n int) cobra.PositionalArgs {
...
+// MinimumNObjectIDArgs returns an error if there is not at least N args and
+// any of those args is not an ObjectID.
+func MinimumNObjectIDArgs(n int) cobra.PositionalArgs {
+	return func(cmd *cobra.Command, args []string) error {
+		if err := MinimumNArgs(n)(cmd, args); err != nil {
+			return err
+		}
+		return objectIDArgs(args)
+	}
+}
```
Comment (andreaangiolillo) [inline internal/cli/require/require.go:105]: [q] same question as `ExactObjectIDArgs`

**Review by andreaangiolillo — APPROVED** [review]
LGTM. Left some questions

File: internal/cli/require/require.go (hunk 1 above)
Comment (gssbzn) [reply internal/cli/require/require.go:63]: At this point that would be a premature abstraction, specially if you mean here is to have something like `ExactCustomValidArgs(n int, f func([]string) error) cobra.PositionalArgs` and specially as you can already do that at the cobra level, we may just have enought of a excuse to make the objectID validation aa thing as many commands will take an objectID. What other custom validations you see here?

Comment (andreaangiolillo) [reply internal/cli/require/require.go:63]: I wasn’t thinking about any particular validation but I was wondering if this change could make sense here. Thanks for the answer!

**Pull request merged**